src/server/process_request.c

```diff
@@ -1410,7 +1514,12 @@ free_br(struct batch_request *preq)
 			break;
 
 #ifndef PBS_MOM		/* Server Only */
-
+		case PBS_BATCH_RegisterSched:
+			if (preq->rq_ind.rq_rsched.rq_name)
+				free(preq->rq_ind.rq_rsched.rq_name);
+			if (preq->rq_ind.rq_rsched.rq_conn_type)
+				free(preq->rq_ind.rq_rsched.rq_conn_type);
```

Comment:
free(NULL) is a noop, there is no need to check if there is anything in them first.